Week1/assignments/armand-collins/Dominator.js

```diff
@@ -0,0 +1,29 @@
+
+
+const dominator = (arr) => {
+  let values = {};
+  let maxValue = arr.length/2
+  let dominator;
+
+  for(let val of arr) {
+    if(values[val] == null) {
+      values[val] = 1;
+    } else {
+      values[val]++;
+    }
+  }
+
+  for(let count of arr) {
+    if(values[count] > maxValue) {
+      dominator = count;
+    }
+  }
+  
+  if(dominator) {
+    return dominator;
+  }
+  return 'It doesnt exist'
+}
+
+console.log(dominator([3, 4, 2, 3, 3, 2, -1, 3, 3]));
```

Comment:
@Fogha your function doesn't return the required values.

The question clearly states: `returns the index of any element of array A in which the dominator of A occurs. The function should return −1 if array A does not have a dominator.`